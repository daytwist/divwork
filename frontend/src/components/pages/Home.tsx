import { FC, useContext } from "react";
import { useNavigate, Link } from "react-router-dom";
import Cookies from "js-cookie";
import { Button, Typography } from "@mui/material";
import Grid2 from "@mui/material/Unstable_Grid2/Grid2";
import { AxiosRequestConfig, AxiosResponse } from "axios";
import { axiosInstance } from "../../utils/axios";
import { AuthResponse } from "../../types";
import { AuthContext } from "../../providers/AuthProvider";
import { SnackbarContext } from "../../providers/SnackbarProvider";

const Home: FC = () => {
  const { setIsSignedIn } = useContext(AuthContext);
  const { handleSetSnackbar } = useContext(SnackbarContext);
  const navigate = useNavigate();

  const handleGuestSignIn = () => {
    const options: AxiosRequestConfig = {
      url: "/auth/guest_sign_in",
      method: "POST",
    };

    axiosInstance(options)
      .then((res: AxiosResponse<AuthResponse>) => {
        console.log(res);
        Cookies.set("_access_token", res.headers["access-token"]);
        Cookies.set("_client", res.headers.client);
        Cookies.set("_uid", res.headers.uid);
        setIsSignedIn(true);
        handleSetSnackbar({
          open: true,
          type: "success",
          message: "ゲストログインしました",
        });
        navigate("/teams", { replace: false });
      })
      .catch((err) => {
        console.log(err);
        handleSetSnackbar({
          open: true,
          type: "error",
          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
          message: `${err.response.data.errors}`,
        });
      });
  };

  return (
    <div>
      <Typography
        variant="h2"
        component="div"
        gutterBottom
        data-testid="home-title"
      >
        DivWork
      </Typography>
      <Grid2 container direction="column" rowSpacing={2}>
        <Grid2 xs={12}>
          <Button
            variant="contained"
            type="button"
            component={Link}
            to="/sign_up/teams/select"
          >
            ユーザー登録
          </Button>
        </Grid2>
        <Grid2 xs={12}>
          <Button
            color="secondary"
            variant="contained"
            type="button"
            component={Link}
            to="/sign_in"
          >
            ログイン
          </Button>
        </Grid2>
        <Grid2 xs={12}>
          <Button color="secondary" type="button" onClick={handleGuestSignIn}>
            ゲストユーザーでログイン
          </Button>
        </Grid2>
      </Grid2>
    </div>
  );
};

export default Home;
