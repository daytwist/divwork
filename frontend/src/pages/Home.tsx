import { FC, useContext } from "react";
import { useNavigate } from "react-router-dom";
import Cookies from "js-cookie";
import { Button, Container, Grid, Typography } from "@mui/material";
import { AxiosRequestConfig, AxiosResponse } from "axios";
import { axiosInstance } from "../utils/axios";
import { AuthResponse } from "../types";
import { AuthContext } from "../providers/AuthProvider";

const Home: FC = () => {
  const { setIsSignedIn } = useContext(AuthContext);
  const navigate = useNavigate();

  const handleGuestSignIn = () => {
    const options: AxiosRequestConfig = {
      url: "/auth/guest_sign_in",
      method: "POST",
    };

    axiosInstance(options)
      .then((res: AxiosResponse<AuthResponse>) => {
        console.log(res);

        if (res.status === 200) {
          Cookies.set("_access_token", res.headers["access-token"]);
          Cookies.set("_client", res.headers.client);
          Cookies.set("_uid", res.headers.uid);

          setIsSignedIn(true);
          navigate(`/teams/${res.data.data.team_id}`, { replace: false });
        }
      })
      .catch((err) => console.log(err));
  };

  return (
    <Container maxWidth="sm">
      <Typography variant="h2" component="div" gutterBottom>
        DivWork
      </Typography>
      <Grid container direction="column" spacing={2}>
        <Grid item>
          <Button
            variant="contained"
            type="button"
            href="/sign_up/teams/select"
          >
            サインアップ
          </Button>
        </Grid>
        <Grid item>
          <Button variant="outlined" type="button" href="/sign_in">
            サインイン
          </Button>
        </Grid>
        <Grid item>
          <Button color="secondary" type="button" onClick={handleGuestSignIn}>
            ゲストユーザーとしてログイン
          </Button>
        </Grid>
      </Grid>
    </Container>
  );
};

export default Home;
