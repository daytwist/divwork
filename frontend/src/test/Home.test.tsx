import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from "../pages/Home";

test("renders learn react link", () => {
  render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );
  const linkElement = screen.getByText(/DivWork/i);
  expect(linkElement).toBeInTheDocument();
});
