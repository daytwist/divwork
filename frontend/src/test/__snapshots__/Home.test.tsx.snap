// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home スナップショット 1`] = `
<div
  className="Home"
>
  <h1>
    DivWork
  </h1>
  <div>
    <a
      href="/sign_up/teams/select"
      onClick={[Function]}
    >
      サインアップ
    </a>
  </div>
  <div>
    <a
      href="/sign_in"
      onClick={[Function]}
    >
      サインイン
    </a>
  </div>
</div>
`;
